Add nits around auth, Codexec, and Codex before DASH (#37280)
DataDog/documentation: content/en/mcp_server/setup.md

<div class="alert alert-danger">Datadog MCP Server is not supported for your selected site ({{< region-param key="dd_site_name" >}}).</div>
{{< /site-region >}}

> <div class="alert alert-info">The <a href="https://github.com/openai/plugins/tree/main/plugins/datadog">Codex Plugin (Preview)</a> can be used in the Codex Desktop app in the US1 regions only. To install, use the ChatGPT App instructions. Once the ChatGPT app is installed, the Codex Plugin should be automatically included as well.  

Reviewer: Why link to the plugin repo? It doesn't seem like users can do anything from there; they'll still need to go through the ChatGPT app to enable the plugin.

I can add a link to the ChatGPT setup instructions in follow-up PR https://github.com/DataDog/documentation/pull/37294. 

```suggestion
<div class="alert alert-info">The <a href="https://github.com/openai/plugins/tree/main/plugins/datadog">Codex Plugin (Preview)</a> can be used in the Codex Desktop app in the US1 regions only. To install, use the ChatGPT App instructions. Once you install the ChatGPT app, the Codex Plugin is automatically included as well.  
```

For security, use a scoped API key and application key from a [service account][13] that has only the required permissions.

> ### Adding OAuth Clients

Reviewer: ```suggestion
### Adding OAuth clients
```

You can allow-list your redirect URLs in [Organization Preferences][27] under `MCP OAuth Redirect URLs`. 

> If you are a partner or vendor adding Datadog to an MCP Directory for your AI Agent platform, please submit your interest through our [Technology Partner Signup][58]

Reviewer: ```suggestion
If you are a partner or vendor adding Datadog to an MCP directory for your AI agent platform, submit your interest through Datadog's [Technology Partner Signup][58].
```
